src/App.js

```diff
@@ -1,84 +1,99 @@
+import AddTodoInput from './component/AddTodoInput';
+import Header from './component/Header';
+import Tabs from './component/Tabs';
 
 class App extends Component {
+
   constructor() {
     super();
     this.state = {
-      todoList: [
-        {
-          id: 123,
-          content: 'create react app',
-          completed: false
-        },
-        {
-          id: 124,
-          content: 'create java app',
-          completed: true
-        }
-      ],
+      todoList: [],
       currentView: 'home'
     };
   }
 
   render() {
-    let currentView = this.state.currentView;
-
     return (
-      <div>
-        {/* Header */}
-        <div>
-          {/* tabs */}
-          <div>
-            <ul>
-              <li onClick={() => this.setState({ currentView: 'home' })}>
-                Home
-              </li>
-              <li onClick={() => this.setState({ currentView: 'completed' })}>
-                Completed
-              </li>
-              <li onClick={() => this.setState({ currentView: 'remaining' })}>
-                Remaining
-              </li>
-            </ul>
-          </div>
+      <div className='main-container__div'>
+        <div className='header_container__div'>
+          {/* Header */}
+          <Header />
+
+          {/* Tabs */}
+          <Tabs setCurrentView={this.setCurrentView} />
+
+          {/* Body */}
+          <AddTodoInput addTodoItem={this.addTodoItem} />
         </div>
 
-        {/* Body */}
-        {currentView === 'home' && (
-          <Home
-            addTodoItem={this.addTodoItem}
-            todoList={this.state.todoList}
+        <div className='div_container__div'>
+          {/* Body */}
+          <Container
+            todoList={this.getTodoProps()}
             toggleTodoItemCompleted={this.toggleTodoItemCompleted}
             deleteTodoItem={this.deleteTodoItem}
+            editTodoItem={this.editTodoItem}
           />
-        )}
-        {currentView === 'remaining' && <Remaining />}
-        {currentView === 'completed' && <Completed />}
+        </div>
       </div>
     );
   }
 
+  getTodoProps = () => {
+    let todoList;
+
+    switch (this.state.currentView) {
+      case 'home':
+        todoList = this.state.todoList;
+        break;
+      case 'remaining':
+        todoList = this.getItemList(false);
+        break;
+      case 'completed':
+        todoList = this.getItemList(true);
+        break;
+      default:
+    }
+
+    return todoList;
+  };
+
   addTodoItem = todoContent => {
-    let todoList = this.state.todoList,
-      newId = Date.now(),
-      todo = {
-        id: newId,
-        content: todoContent,
-        completed: false
-      };
+    let newId = Date.now();
 
-    todoList.push(todo);
+    let todo = {
+      id: newId,
+      content: todoContent,
+      completed: false
+    };
 
-    this.setState(todoList);
+    let todoList = [...this.state.todoList, todo];
+    this.setState({ todoList });
```

Comment:
We are not changing the values of `todo` and `todoList` variables so let's use `const` instead of `let`